homeassistant/components/knx/trigger.py

```diff
@@ -0,0 +1,104 @@
+"""Offer knx telegram automation triggers."""
+
+from typing import Final
+
+import voluptuous as vol
+from xknx.telegram.address import DeviceGroupAddress, parse_device_group_address
+
+from homeassistant.const import CONF_PLATFORM
+from homeassistant.core import CALLBACK_TYPE, HassJob, HomeAssistant, callback
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.trigger import TriggerActionType, TriggerInfo
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DOMAIN, SIGNAL_KNX_TELEGRAM_DICT
+from .schema import ga_validator
+from .telegrams import TelegramDict
+
+TRIGGER_TELEGRAM: Final = "telegram"
+
+PLATFORM_TYPE_TRIGGER_TELEGRAM = f"{DOMAIN}.{TRIGGER_TELEGRAM}"
+
+CONF_KNX_DESTINATION = "destination"
+CONF_KNX_GROUP_VALUE_WRITE = "group_value_write"
+CONF_KNX_GROUP_VALUE_READ = "group_value_read"
+CONF_KNX_GROUP_VALUE_RESPONSE = "group_value_response"
+CONF_KNX_INCOMING = "incoming"
+CONF_KNX_OUTGOING = "outgoing"
+
+TELEGRAM_TRIGGER_SCHEMA = {
+    vol.Optional(CONF_KNX_DESTINATION): vol.Maybe(
```

Comment:
Why do we need allow `None` here?

Reply:
See the PR description. There is also a blueprint example for this.
> If destination is explicitly set to None, the trigger is a no-op. This can be used in blueprints to provide inputs for addresses that are optional. With an input with default: null the trigger will never fire if no address is provided.

https://github.com/home-assistant/core/pull/114458 could supersede this.

Reply:
OK

Reply:
If we could find a decision about the linked PR, this `None` handling option could be removed completely without further (breaking)changes or deprecations to this trigger to accomplish the same goal for blueprints.